internal/testrunner/runners/system/tester.go

```diff
@@ -456,11 +456,18 @@ func (r *tester) createAgentInfo(policy *kibana.Policy, config *testConfig, runI
 	info.Agent.AgentSettings = config.Agent.AgentSettings
 
 	// If user is defined in the configuration file, it has preference
-	// and it should not be overwritten by the value in the manifest
-	if info.Agent.User == "" && agentManifest.Privileges.Root {
+	// and it should not be overwritten by the value in the package or DataStream manifest
+	if info.Agent.User == "" && (r.pkgManifest.Agent.Privileges.Root || r.dataStreamManifest.Agent.Privileges.Root) {
```

Comment:
Previously, it was just considered the privileges settings from package manifest. It was no taken into account the one from the data stream manifest.